omod/src/main/java/org/openmrs/module/coreapps/page/controller/MergeVisitsPageController.java

```diff
@@ -52,27 +64,58 @@ public Object controller(@RequestParam("patientId") Patient patient,
 
     public String post(@RequestParam("patientId") Patient patient,
                        @RequestParam("mergeVisits") List<Integer> mergeVisits,
-					   @RequestParam(value = "returnUrl", required = false) String returnUrl,
+                       @RequestParam(value = "returnUrl", required = false) String returnUrl,
                        @SpringBean AdtService service,
                        UiUtils ui,
-                       HttpServletRequest request, PageModel model){
-
+                       HttpServletRequest request, PageModel model) {
+        List<String> mergedVisitsIDs = new ArrayList<String>();
         if (patient.isVoided() || patient.isPersonVoided()) {
             return "redirect:" + ui.pageLink("coreapps", "patientdashboard/deletedPatient",
-					SimpleObject.create("patientId", patient.getId().toString() ));
+                    SimpleObject.create("patientId", patient.getId().toString()));
         }
 
-        if (mergeVisits!=null && mergeVisits.size() > 0 ){
+        if (mergeVisits != null && mergeVisits.size() > 0) {
             Visit mergedVisit = service.mergeConsecutiveVisits(mergeVisits, patient);
-            if (mergedVisit != null){
+            if (mergedVisit != null) {
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE, ui.message("coreapps.task.mergeVisits.success"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
-            }else{
+            } else {
                 request.getSession().setAttribute("emr.errorMessage", ui.message("coreapps.task.mergeVisits.error"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
             }
+
+            for (int i = 1; i < mergeVisits.size(); i++) {
+                mergedVisitsIDs.add(mergeVisits.get(i).toString());
+            }
         }
 
-		return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl));
+
+        return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl, "mergedVisitsIDs", mergedVisitsIDs));
+    }
+
+
+    /**
+     * This method receives the returnUrl and the list of the merged visit IDs
+     * If some of the merged visit IDs are on the return url, it will remove that parameter from the url
+     * so it doesn't return to a visit that no longer exists.
+     * @param returnUrl the return URL.
+     * @param mergedVisitsIDs list of voided visits that have been merged.
+     * @return The return URL, without the visit param if that visit was merged.
+     */
+    public String updateURLWithoutMergedVisits(String returnUrl, List<String> mergedVisitsIDs) {
```

Comment:
Maybe `updateVisitIdUrlParam` or `updateVisitIdUrlParameter`:
```java
public String updateVisitIdUrlParam(String url, List<String> visitIds) {
```
This is just taking 1) an URL and 2) a list of visit IDs. Then, if the URL's `visitId` param's value is one of the visit IDs from the list, it scraps the `visitId` param altogether off the URL. For the sake of this method there is no need to speak about return URL or merged visit IDs.

Reply:
I will change it to `updateVisitIdUrlParameter`.